specification/ml/_types/TrainedModel.ts

```diff
@@ -372,19 +421,19 @@ export class TrainedModelAssignmentRoutingTable {
    * The reason for the current state. It is usually populated only when the
    * `routing_state` is `failed`.
    */
-  reason: string
+  reason?: string
   /**
    * The current routing state.
    */
   routing_state: RoutingState
   /**
    * Current number of allocations.
    */
-  current_allocations: integer
+  current_allocations?: integer
   /**
    * Target number of allocations.
    */
-  target_allocations: integer
+  target_allocations?: integer
```

Comment:
Those are not optional starting from 8.4:

https://github.com/elastic/elasticsearch/blob/045f6a31f994f51d87a217be60251e060132c8a1/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/RoutingInfo.java#L134-L135